functions/Get-DbaDbFile.ps1

```diff
@@ -19,6 +19,9 @@ function Get-DbaDbFile {
     .PARAMETER ExcludeDatabase
         The database(s) to exclude - this list is auto-populated from the server
 
+    .PARAMETER SmoDatabase
+        A SMO Database object that represents the database to reference
```

Comment:
This is the intent of InputObject: InputObject should be the SMODatabase.

`[Microsoft.SqlServer.Management.Smo.Database[]]$InputObject,` You can do that, and then InputObject +=.